embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/KafkaDataFormatsTest.java

```diff
@@ -0,0 +1,716 @@
+import org.apache.druid.data.input.avro.SchemaRegistryBasedAvroBytesDecoder;
+import org.apache.druid.data.input.impl.CsvInputFormat;
+import org.apache.druid.data.input.impl.DelimitedInputFormat;
+import org.apache.druid.data.input.impl.DimensionsSpec;
+import org.apache.druid.data.input.impl.JsonInputFormat;
+import org.apache.druid.data.input.impl.TimestampSpec;
+import org.apache.druid.data.input.protobuf.FileBasedProtobufBytesDecoder;
+import org.apache.druid.data.input.protobuf.ProtobufExtensionsModule;
+import org.apache.druid.data.input.protobuf.ProtobufInputFormat;
+import org.apache.druid.data.input.protobuf.SchemaRegistryBasedProtobufBytesDecoder;
+import org.apache.druid.indexing.kafka.KafkaIndexTaskModule;
+import org.apache.druid.indexing.kafka.simulate.KafkaResource;
+import org.apache.druid.indexing.kafka.supervisor.KafkaSupervisorIOConfig;
+import org.apache.druid.indexing.kafka.supervisor.KafkaSupervisorSpec;
+import org.apache.druid.indexing.kafka.supervisor.KafkaSupervisorTuningConfig;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.DruidMetrics;
+import org.apache.druid.segment.TestHelper;
+import org.apache.druid.segment.indexing.DataSchema;
+import org.apache.druid.testing.embedded.EmbeddedBroker;
+import org.apache.druid.testing.embedded.EmbeddedCoordinator;
+import org.apache.druid.testing.embedded.EmbeddedDruidCluster;
+import org.apache.druid.testing.embedded.EmbeddedHistorical;
+import org.apache.druid.testing.embedded.EmbeddedIndexer;
+import org.apache.druid.testing.embedded.EmbeddedOverlord;
+import org.apache.druid.testing.embedded.junit5.EmbeddedClusterTestBase;
+import org.apache.druid.testing.tools.AvroSchemaRegistryEventSerializer;
+import org.apache.druid.testing.tools.EventSerializer;
+import org.apache.druid.testing.tools.ProtobufSchemaRegistryEventSerializer;
+import org.apache.druid.testing.tools.StreamGenerator;
+import org.apache.druid.testing.tools.WikipediaStreamEventStreamGenerator;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.joda.time.Period;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Embedded test of kafka indexing for different data formats.
+ * Formats Included:
+ * <ul>
+ * Avro (with and without schema registry)
+ * CSV
+ * JSON
+ * Protobuf (with and without schema registry)
+ * </ul>
+ *
+ * This tests both InputFormat and Parser. Parser is deprecated for Kafka Ingestion, and those tests will be removed in the future.
+ */
+public class KafkaDataFormatsTest extends EmbeddedClusterTestBase
+{
+  private static final long CYCLE_PADDING_MS = 100;
+  private static final int EVENTS_PER_SECOND = 6;
+
+
+  private static ObjectMapper jsonMapper;
```

Comment:
This can be a constant or you can use the mapper bindings from one of the servers, say:
```
overlord.bindings().jsonMapper()
```